### ISSUE TITLE
Feature request: Kotlin optimization

### ISSUE THREAD

**Issue by MukjepScarlet** [vigna/fastutil#339]
1. Some APIs will conflict in Kotlin.
Example:
```kotlin
Long2ObjectMap.computeIfAbsent(long, V) <-> Long2ObjectMap.computeIfAbsent(java.lang.Long, V)
```

2. Kotlin built-in extensions will make many classes degenerate to normal type.
Example:

`Object2LongMap.forEach` will produce entry with type `java.lang.Object` and `java.lang.Long`. And `Map.put(K, V)`, etc.

`Entry<K,V>.component1` / `component2`

One of the resolutions: [PairExtensions.kt](https://github.com/CCBlueX/LiquidBounce/blob/nextgen/src/main/kotlin/net/ccbluex/liquidbounce/utils/kotlin/PairExtensions.kt)
I hope these stuffs could be library built-in ones, so Kotlin will give priority to existing member methods rather than extension functions

**Comment by vigna** [comment]
Well, I don't understand exactly what you're asking for here—note that I'm aware of Kotlin, but I have zero knowledge of the language...

**Comment by MukjepScarlet** [comment]
Well, I mean when you use this library in a Kotlin project, some frustrating conflicts will make the code more complicated.
In Kotlin we iterate a map like:
```kotlin
myMap.forEach { (key, value) ->
    ...something
} // key has type K and value has type V (Map.Entry<K,V>)
```
(`componentN` allow us to use destructuring expressions like `var (key, value) = entry`)

If `myMap` is an `Int2ObjectOpenHashMap`, and we don't want to let the key become an `Integer`, we have to:
```kotlin
myMap.int2ObjectEntrySet().forEach { entry ->
    val key = entry.intKey
    val value = entry.value
    ...something
}
```
So I hope the library can have built-in `component1` and `component2` functions for entries and pairs

**Comment by incaseoftrouble** [comment]
So to add that you would need public functions `key_type component1()` and `value_type component2()` on the Map.Entry?

**Comment by MukjepScarlet** [comment]
Yeah, but type-specified ones, like the `int intValue()` and maybe `@Deprecated Integer value()`

**Comment by incaseoftrouble** [comment]
Hmmm well that would pollute the autocompletion for all Java users. I think the better solution would be to provide a primitive variant of `forEach` (using appropriate BiConsumers). Something like `forEachPrimitive(IntDoubleBiConsumer action)` etc.

**Comment by MukjepScarlet** [comment]
I think it might be better to create a "Kotlin module" of this lib like what Jackson does, providing extensions and utils.
